nautobot/project-static/js/forms.js

```diff
@@ -50,13 +50,14 @@ function colorPickerClassCopy(data, container) {
 */
 
 // Static choice selection
-function initializeStaticChoiceSelection(context){
+function initializeStaticChoiceSelection(context, dropdownParent=null){
```

Comment:
Instead of manually specifying a `dropdownParent` (and reinitializing menus inside a modal a second time), could we do something more automatic, like (not tested):

```js
this_context.find('.nautobot-select2-static').each(function() {
    $(this).select2({
        ...
        dropdownParent: $(this).closest(".modal")[0],
    });
)}
```

Reply:
Since there are many more select widget that aren't in modals I thought the approach you were suggesting would require more iterations and checks. Reinitializing seems to be no issue for Select2 and would be a much smaller subset of widgets (since the call inwards would be scoped to the modal's context).